### PR TITLE
integrate: small cleanup in Simpsons

### PR DESCRIPTION
**Pull request by vladimir-ch**
A follow-up to #1133 

Please take a look.

### CODE REVIEW

File: integrate/simpsons.go [hunk 1]
```diff
@@ -27,31 +27,6 @@ func Simpsons(x, f []float64) float64 {
 		panic("integrate: must be sorted")
 	}
 
-	// Simpson's method approximates the integral of a function f by subdividing
```
Comment (vladimir-ch) [inline integrate/simpsons.go]: In the end I decided to delete this comment. It just repeats what's already in the code and moreover the code is probably clearer.

File: integrate/simpsons_test.go [hunk 2]
```diff
@@ -21,20 +21,24 @@ func TestSimpsons(t *testing.T) {
 		tol      float64
 	}{
 		{integral: testquad.Constant(0), n: 3, tol: 0},
+		{integral: testquad.Constant(0), n: 4, tol: 0},
```
Comment (vladimir-ch) [inline integrate/simpsons_test.go]: The intention is to have a test with the minimum even number of nodes.

File: integrate/simpsons_test.go [hunk 3]
```diff
@@ -21,20 +21,24 @@ func TestSimpsons(t *testing.T) {
...
-		{integral: testquad.Poly(3), n: 1e3, tol: 1e-8},
-		{integral: testquad.Poly(4), n: 1e3, tol: 1e-8},
-		{integral: testquad.Poly(5), n: 1e3, tol: 1e-7},
-		{integral: testquad.Sin(), n: 1e2, tol: 1e-8},
-		{integral: testquad.XExpMinusX(), n: 1e2, tol: 1e-8},
+		{integral: testquad.Poly(3), n: 1000, tol: 1e-8},
+		{integral: testquad.Poly(4), n: 1000, tol: 1e-8},
+		{integral: testquad.Poly(5), n: 1000, tol: 1e-7},
+		{integral: testquad.Sin(), n: 100, tol: 1e-8},
+		{integral: testquad.XExpMinusX(), n: 101, tol: 1e-8},
```
Comment (vladimir-ch) [inline integrate/simpsons_test.go]: This test was failing in my virtual machine, plus there will be a test case with a large odd number of nodes.

Comment (kortschak) [reply integrate/simpsons_test.go]: Also failing on travis and appveyor:
```
--- FAIL: TestSimpsons (0.00s)
    simpsons_test.go:58: Test #16: ∫_0^1 x*exp(-x)dx, n=101: unexpected result; got=0.26424112862624427 want=0.26424111765711533 diff=1.096912893849833e-08
```

Comment (vladimir-ch) [reply integrate/simpsons_test.go]: Yes, I noticed it too late. Let's see if it passes now.

**Pull request merged**